src/classes/groundplaces.ts

```diff
@@ -0,0 +1,131 @@
+
+import { GroundPlace, StopGroup, StopCluster, Properties } from '../types';
+
+interface Filters {}
+
+interface GroundPlacesDiff {}
+
+/**
+ * @description GroundPlaces business logic.
+ */
+export class GroundPlaces {
+
+  /**
+   * @description Returns a list of places.
+   * @param query Can be a name, a Gpuid, a unique name or other name.
+   * @param filters Filters with different options (StopGroup, StopCluster, Serviced, SegmentProvider).
+   * @returns {Array<StopGroup|StopCluster>}
+   */
+  // @ts-ignore
+  public autocomplete(query: string, filters: Filters): (StopGroup | StopCluster)[] {}
+
+  /**
+   * @description Create stopCluster or stopGroup.
+   * @param type Type can be stopCluster or stopGroup.
+   * @param groundPlace Informations to use for creating the place.
+   * @param GpuidParent Ground place unique identifier of the stopCluster parent if the type is a stopGroup. 
+   * @returns {StopGroup|StopCluster}
+   */
+  // @ts-ignore
+  // This method will call Storage.createStopGroup or Storage.createStopCluster depending on type passed in params.
+  public create(type: 'stopCluster' | 'stopGroup', groundPlace: GroundPlace, GpuidParent?: string): StopGroup | StopCluster {}
+
+  /**
+   * @description Create stopCluster or stopGroup.
+   * @param type Type can be stopCluster or stopGroup.
+   * @param Gpuid Ground place unique identifier.
+   * @param propertiesToUpdate Properties that need to be update. 
+   * @returns {StopGroup|StopCluster}
+   */
+  // @ts-ignore
+  // This method will call Store.updateStopGroup or Storage.updateStopCluster depending on type passed in params.
+  public update(type: 'stopCluster' | 'stopGroup', Gpuid: string, propertiesToUpdate: Properties): StopGroup | StopCluster {}
+
+  /**
+   * @description Remove a stopGroup from a stopCluster.
+   * @param stopClusterGpuid StopCluster Ground place unique identifier.
+   * @param stopGroupGpuid StopGroup Ground place unique identifier to remove.
+   * @returns {Void}
+   */
+  public removeStopGroupFromStopCluster(stopClusterGpuid: string, stopGroupGpuid: string): void {}
```

Comment:
Créer deux types pour GPUID : stopClusterGpuid & stropGroupGpuid :) 
(ou seulement 1 des deux : gpuid)

Reply:
@rimiti une préférence ?

Reply:
"diviser pour mieux régner" 👍

Reply:
Gpuid est une string, je ne pense pas que TypeScript permette de typer un string en matchant une certaine regex, en tout cas pas à ma connaissance. Du coup je peux créer ces types, mais ils renverront tout les deux des strings au final, ça ira ?

Reply:
> je peux créer ces types, mais ils renverront tout les deux des strings

It's exactly what we want!